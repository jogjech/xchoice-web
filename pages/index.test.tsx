import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Home from "./index";

const mockRouterPush = jest.fn();

jest.mock("next/router", () => ({
  useRouter: () => {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      push: mockRouterPush,
    };
  },
}));

jest.mock("@auth0/auth0-react", () => ({
  useAuth0: () => {
    return {
      isAuthenticated: true,
      user: {
        nickname: "Kevin Wang",
      },
      logout: jest.fn(),
      loginWithRedirect: jest.fn(),
    };
  },
}));

describe("Home page", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Home></Home>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains layout", () => {
    const layout = wrapper.find("Layout");
    expect(layout.exists()).toBeTruthy();
    expect(layout.prop("title")).toBe("X Choice");
  });

  it("contains title", () => {
    const title = wrapper.find(".title");
    expect(title.text()).toBe("Welcome back, Kevin Wang!");
  });

  it("contains button to create survey", () => {
    const button = wrapper.find("Button");
    expect(button.text()).toBe("Create Survey Now");
    button.simulate("click");
    expect(mockRouterPush).toHaveBeenCalledWith("/survey/create");
  });
});
