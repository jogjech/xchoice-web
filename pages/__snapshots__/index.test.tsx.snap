// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div>
  <Layout
    title="X Choice"
  >
    <div
      className="homeContainer"
    >
      <p
        className="title"
      >
        Welcome back, Kevin Wang!
      </p>
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        shape="round"
        size="large"
        type="primary"
      >
        Create Survey Now
      </Button>
    </div>
  </Layout>
</div>
`;
